chain/jsonrpc/src/lib.rs

```diff
@@ -328,16 +332,55 @@ impl JsonRpcHandler {
         match self.client_addr.send(Status { is_health_check: true }).await {
             Ok(Ok(_)) => Ok(Value::Null),
             Ok(Err(err)) => Err(RpcError::new(-32_001, err, None)),
-            Err(_) => Err(RpcError::server_error::<String>(None)),
+            Err(_) => Err(RpcError::server_error::<()>(None)),
         }
     }
 
     pub async fn status(&self) -> Result<Value, RpcError> {
         match self.client_addr.send(Status { is_health_check: false }).await {
             Ok(Ok(result)) => jsonify(Ok(Ok(result))),
             Ok(Err(err)) => Err(RpcError::new(-32_001, err, None)),
-            Err(_) => Err(RpcError::server_error::<String>(None)),
+            Err(_) => Err(RpcError::server_error::<()>(None)),
+        }
+    }
+
+    /// Expose Genesis Config (with internal Runtime Config) without state records to keep the
+    /// output at a reasonable size.
+    ///
+    /// See also `genesis_records` API.
+    pub async fn genesis_config(&self) -> Result<Value, RpcError> {
+        // We cannot skip `records` field GenesisConfig with serde, as it will break `dump_state`
+        // util. Thus, we make a copy of GenesisConfig with empty `records`, and then remove
+        // `records` from the serialized json representation.
+        let mut json_genesis_config_response = jsonify(Ok(Ok(GenesisConfig {
+            records: vec![],
+            ..Clone::clone(&self.genesis_config)
+        })));
+        if let Ok(ref mut json_genesis_config) = json_genesis_config_response {
+            if let Some(json_genesis_config_object) = json_genesis_config.as_object_mut() {
+                json_genesis_config_object.remove("records");
+            }
+        }
+        json_genesis_config_response
+    }
+
+    /// Expose Genesis State Records with pagination.
+    ///
+    /// See also `genesis_config` API.
+    async fn genesis_records(&self, params: Option<Value>) -> Result<Value, RpcError> {
```

Comment:
Do we actually want to expose this through rpc? Seems pretty expensive

Reply:
We need it for Explorer to know which accounts are there, at the very least.